### DISCUSSION TITLE
What is the length of considered text? How to avoid Error Status 400

### DISCUSSION THREAD

**Discussion by SONDLecT** [nhaouari/obsidian-textgenerator-plugin#82] [Q&A] [answered]
Cheers! First of all, this is an amazing plugin - thank you for your work on it! I fairly frequently run into errors as I'm using the text generator to build out a document once the document gets to a certain length. I'm assuming that it is hitting a token limit, because if i select a portion of the text I can still generate fine. I'm wondering: what _is_ the limit for considered text? Is there any way to increase that? Can I change something on OpenAI's side?

**Comment by nhaouari** [comment, marked as answer]
Thank you for your kind words and for using Text Generator! The limit for considered text is determined by the maximum number of tokens that the plugin can take and generate at once. The default token limit is set to 160, but you can change this limit by adjusting the settings in the plugin or by using the "change max tokens" commqnd. You can check out this link https://text-gen.com/set-max-content-size for more information on how to increase the maximum number of tokens.

**Comment by SONDLecT** [comment]
Thank you for the response! Is there an upper limit though? I've set the limit to 10,000 but when I try to press generate on an empty line (which means it should include the entire note as considered text, right?) on a 3800 word note I get an error telling me to check the console. The console error says:
`plugin:obsidian-textgenerator-plugin:32466 Error: Request failed, status 429
    at new t (app.js:1:710073)
    at Aw (app.js:1:710265)
    at app.js:1:710942
    at app.js:1:235505
    at Object.next (app.js:1:235610)
    at a (app.js:1:234349)`

**Reply by SONDLecT** [reply]
I will add it looks like you can only set the tokens to a max of 2048. Does that mean that the considered text can not be longer than, what, 4000 words? 4000 characters? I have some considerably long text and it's very useful to be able to get synthesis/restatements of the entire text - and I also think this limit would impact teh usability of using child notes as considered text as well.
